Makefile

```diff
@@ -4,28 +4,56 @@ PREFIX ?= /usr
 SHARE_DIR = $(DESTDIR)$(PREFIX)/share
 LIB_DIR = $(DESTDIR)$(PREFIX)/lib
 
-.PHONY: install
+.PHONY: install uninstall help
 
 install:
 	@install -Dm755 -t "$(DESTDIR)/etc/grub.d/" 41_snapshots-btrfs
 	@install -Dm644 -t "$(DESTDIR)/etc/default/grub-btrfs/" config
 	@install -Dm644 -t "$(LIB_DIR)/systemd/system/" grub-btrfs.service
 	@install -Dm644 -t "$(LIB_DIR)/systemd/system/" grub-btrfs.path
 	@install -Dm644 -t "$(SHARE_DIR)/licenses/$(PKGNAME)/" LICENSE
-	@install -Dm644 "initramfs/Arch Linux/overlay_snap_ro-install" "$(LIB_DIR)/initcpio/install/grub-btrfs-overlayfs"	# Arch Linux only
-	@install -Dm644 "initramfs/Arch Linux/overlay_snap_ro-hook" "$(LIB_DIR)/initcpio/hooks/grub-btrfs-overlayfs"		# Arch Linux only
+	@# Arch Linux like distros only :
+	@if command -V mkinitcpio >/dev/null 2>&1; then \
+		install -Dm644 "initramfs/Arch Linux/overlay_snap_ro-install" "$(LIB_DIR)/initcpio/install/grub-btrfs-overlayfs"; \
+		install -Dm644 "initramfs/Arch Linux/overlay_snap_ro-hook" "$(LIB_DIR)/initcpio/hooks/grub-btrfs-overlayfs"; \
+	 fi
 	@install -Dm644 -t "$(SHARE_DIR)/doc/$(PKGNAME)/" README.md
 	@install -Dm644 "initramfs/readme.md" "$(SHARE_DIR)/doc/$(PKGNAME)/initramfs-overlayfs.md"
 
 uninstall:
-	rm -f "$(DESTDIR)/etc/grub.d/41_snapshots-btrfs"
-	rm -f "$(DESTDIR)/etc/default/grub-btrfs/config"
-	rm -f "$(LIB_DIR)/systemd/system/grub-btrfs.service"
-	rm -f "$(LIB_DIR)/systemd/system/grub-btrfs.path"
-	rm -f "$(SHARE_DIR)/licenses/$(PKGNAME)/LICENSE"
-	rm -f "$(DESTDIR)/boot/grub/grub-btrfs.cfg"
-	rm -f "$(LIB_DIR)/initcpio/install/grub-btrfs-overlayfs" # Arch Linux only
-	rm -f "$(LIB_DIR)/initcpio/hooks/grub-btrfs-overlayfs" # Arch Linux only
-	rm -rf "$(SHARE_DIR)/doc/$(PKGNAME)/" README.md
-	rm -rf "$(SHARE_DIR)/doc/$(PKGNAME)/initramfs-overlayfs.md"
-	rmdir --ignore-fail-on-non-empty "$(DESTDIR)/etc/default/grub-btrfs"
+	@rm -f "$(DESTDIR)/etc/grub.d/41_snapshots-btrfs"
+	@rm -f "$(DESTDIR)/etc/default/grub-btrfs/config"
+	@rm -f "$(LIB_DIR)/systemd/system/grub-btrfs.service"
+	@rm -f "$(LIB_DIR)/systemd/system/grub-btrfs.path"
+	@rm -f "$(DESTDIR)/boot/grub/grub-btrfs.cfg"
```

Comment:
`grub-btrfs.cfg` isn't included during installation, this file is generated by the main script and its destination path is configurable via the config file.
The idea would be to read the config file (before deleting `/etc/default/grub-btrfs/config`) to delete this file correctly.

Maybe like this:
```
@grub_dirname=$$(grep -oP '^[[:space:]]*GRUB_BTRFS_GRUB_DIRNAME="\K.+[^"]' /etc/default/grub-btrfs/config); \
rm -f "$$grub_dirname/grub-btrfs.cfg"
```
`grep` command should return value of the `GRUB_BTRFS_GRUB_DIRNAME` in config file.

Reply:
Yo, nice one !

The grep regex won't work though if the path isn't double quoted in the config file (it could also be single quoted or unquoted). What about simply evaluating the relevant line and using the original variable instead (with a default value of `/boot/grub` in case it is unset) : 

```sh
	@eval "$$(grep '^\s*GRUB_BTRFS_GRUB_DIRNAME=' "$(DESTDIR)/etc/default/grub-btrfs/config")"; \
	 rm -f "$(DESTDIR)$${GRUB_BTRFS_GRUB_DIRNAME:-/boot/grub}/grub-btrfs.cfg"
```

Reply:
I will never use the eval command, with administrator rights on an unprotected variable that can be modified by the user.

Reply:
Agreed. I'll just use grep then.

Reply:
```sh
@grub_dirname="$$(grep -oP '^[[:space:]]*GRUB_BTRFS_GRUB_DIRNAME=\K.*' "$(DESTDIR)/etc/default/grub-btrfs/config")"; \
 rm -f "$(DESTDIR)$${grub_dirname:-/boot/grub}/grub-btrfs.cfg"
```
?

Reply:
This fails when the config = `"/boot/grub"` as define per default in the conf file
`'"/boot/grub"/grub-btrfs.cfg'`

Why used `$(DESTDIR)` here :
` rm -f "$(DESTDIR)$${grub_dirname:-/boot/grub}/grub-btrfs.cfg"`
?

Reply:
> Why used $(DESTDIR) here

Good question, that doesn't make sense. I used that for testing (installing/uninstalling to `/tmp/somefolder`.

> This fails when the config = "/boot/grub" as define per default in the conf file

I'm trying to eliminate quotes and comments with shell variable substitution and/or `sed`.

Reply:
>  I used that for testing

Ok :)

> I'm trying to eliminate quotes and comments

A idea ? `sed -e 's|["'\''()]||g; s/#.*//' <<< ('"/boot/grub"'     # comment)`

Reply:
That looks good. I'll try that ! :)

Reply:
So I guess (hope) I got something sensible now : 

```sh
@grub_dirname="$$(grep -oP '^[[:space:]]*GRUB_BTRFS_GRUB_DIRNAME=\K.*' "$(DESTDIR)/etc/default/grub-btrfs/config" | sed "s|['\"]||g;s|\s*#.*||")"; \
 rm -f "$${grub_dirname:-/boot/grub}/grub-btrfs.cfg"
```

This works with both single and double quotes, as well as unquoted path. Tested also paths containing spaces, everything with and without comments.

Reply:
So I've pushed a new commit, which should now do the task. This has been tested with the following cases : 

```sh
#GRUB_BTRFS_GRUB_DIRNAME=/boot/grub2
GRUB_BTRFS_GRUB_DIRNAME="/boot/grub2"
GRUB_BTRFS_GRUB_DIRNAME="/boot/grub2"   # comment
GRUB_BTRFS_GRUB_DIRNAME="/boot/grub2"# comment
GRUB_BTRFS_GRUB_DIRNAME='/boot/grub2'
GRUB_BTRFS_GRUB_DIRNAME='/boot/grub 2'  # comment
GRUB_BTRFS_GRUB_DIRNAME=/boot/grub2		# comment  # hash sign inside comment
GRUB_BTRFS_GRUB_DIRNAME=("/boot/grub2")
GRUB_BTRFS_GRUB_DIRNAME=('/boot/grub2')
GRUB_BTRFS_GRUB_DIRNAME=(/boot/grub2)
GRUB_BTRFS_GRUB_DIRNAME=( /boot/grub2 )
GRUB_BTRFS_GRUB_DIRNAME=(	'/boot/grub2'	)	# (Tabs)
GRUB_BTRFS_GRUB_DIRNAME=(	'/whitespace/grub 2'	)	# comment
```

I think these are the important cases that must be handled. Please let me know if I've missed something that isn't covered yet.